src/mlang/optimizing_ir/partial_evaluation.ml

```diff
@@ -393,12 +381,13 @@ let rec partially_evaluate_expr (ctx : partial_ev_ctx) (p : Mir.program)
               from_literal (Float 0.)
           (* division *)
           | Mast.Div, (e', _), (Literal (Float 1.), _) -> (e', d1)
+          | Mast.Div, (_, Float), (Literal (Float 0.), _)
           | Mast.Div, (Literal (Float 0.), _), (_, Float) ->
               from_literal (Mir.Float 0.)
           (* default case *)
-          | (Mast.Add | Mast.Sub), _, _ ->
+          | (Mast.Add | Mast.Sub | Mast.Or), _, _ ->
```

Comment:
??? `Or` has a difference behavior than `And` ???

Reply:
Yes, same difference between `+` and `*`. Or at least code generation works that way and the tests don't seem to mind.

Reply:
Wouahou well spotted.

Reply:
Has this been fixed in the interpreter as well ?

Reply:
It already was there since forever, apparently